### ISSUE TITLE
Incompatibility between Loginpass/FastAPI and Authlib

### ISSUE THREAD

**Issue by watdev2018** [authlib/loginpass#77]
In Loginpass there is a common function used to normalize user information. For example, https://github.com/authlib/loginpass/blob/master/loginpass/twitter.py#L17 Notice that it is not an async function:
`def normalize_userinfo(client, data):`

However, there is an attempt to call it as async in https://github.com/lepture/authlib/blob/master/authlib/integrations/base_client/async_app.py#L142

The incompatibility triggers an error.

What is the best way to address this? Currently I'm removing the `await` for my own purpose but I don't know the impact of this

**Comment by boertel** [comment]
I came across the same error. My workaround is to make these `normalize_userinfo` functions asynchronous. To do that I'm overwriting my connections like:
```python
from loginpass import Twitter as LPTwitter, normalize_userinfo
from asgiref.sync import sync_to_async

OAUTH_CONFIG = LPTwitter.OAUTH_CONFIG
OAUTH_CONFIG["userinfo_compliance_fix"] = sync_to_async(normalize_userinfo)

class Twitter(LPTwitter):
   OAUTH_CONFIG = OAUTH_CONFIG
```

I found out about this `sync_to_async` method in the django docs: https://docs.djangoproject.com/en/3.1/topics/async/#sync-to-async.

You still have to re-define all your connections but at least there is no need to update `authlib` directly and the connections would still rely on most of `loginpass` definitions. If you have a lot of connections, I would suggest having a function: `Twitter = make_it_asynchronous(loginpass.Twitter)` to do the work for you 😄
